src/snowflake/snowpark/mock/_plan.py

```diff
@@ -394,11 +405,21 @@ def handle_function_expression(
             raise_error=NotImplementedError,
         )
 
+    # Use the non-decorated function.
+    if "__wrapped__" in original_func.__dict__:
+        original_func = original_func.__wrapped__
```

Comment:
this comes from the publicapi decorator. Basically, for something like when using `@udf` or so together with `publicapi` the signature extraction code below won't pick up the signature from the underlying function. This can be simply fixed by relying on the `__wrapped__` attribute.